### ISSUE TITLE
podman ps -a: unable to create output: container [...] does not exist in database

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#418]
On a busy system running docker-autotest, I manually ran `podman ps -a` and got:

```
# podman ps -a 
unable to create output: container c516336ee82e216ff3be5afc8b17324d57a12d8557fd3925eab87cb325b978f1 does not exist in database: no such container                    
```
I can reproduce in a few seconds via:

```
shell1# while :;do podman run --rm registry.access.redhat.com/rhel7/rhel:latest true;done

shell2# while :;do podman ps -a;done      ! In >10 minutes of testing, I was unable to reproduce with just 'ps' (no -a)
```

Looks like a race condition but I'm darned if I can figure out where. And I have the sinking feeling that this might require a lock to resolve; I hope I'm wrong and there's an easier way.

**Comment by rhatdan** [comment]
I would rather just eat the error, no locks please.  I think ps is getting a list of containers and is then going to each container to get more information.  If the call to get info returns that the container does not exist, we should just continue and not return the error message.

**Comment by mheon** [comment]
So this is actually the result of race condition avoidance in `libpod` causing another race condition in `podman`.

`podman ps` currently grabs a list of all containers from `libpod` and then iterates through them individually, getting stats. We do not hold locks between the retrieving and iterating. As such, containers can be removed before we get to them in `ps`, causing this error as we try and find them again and they aren't there.

**Comment by mheon** [comment]
We can probably handle this gracefully by just ignoring containers that return `ErrNoSuchCtr` when we go to do our batched operations.

**Comment by edsantiago** [comment]
If this error can be ignored in podman, I'll second everyone's plan to do so. I just got confused because I saw the error coming from libpod `UpdateContainer()` and my gut told me it shouldn't be ignored there.

**Comment by baude** [comment]
I have also seen this when we have unusual containers in the file system but no in the db.  Ive actually seen it quite a bit.,

**Comment by rhatdan** [comment]
Well if we ever merge libpod into crio and buildah hopefully we won't be in that situation any longer.  But for now, lets just check for the ErrNoSuchCtr and don't report it.

**Comment by umohnani8** [comment]
Fixed in #467.
